src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,184 @@
+            }
+        } catch (SQLException e) {
+            logger.error("Can't get user by ID", e);
+        }
+        return Optional.of(user);
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE " + DB_NAME + ".users "
+                + "SET name = ?, surname = ?, login = ?, password = ? "
+                + "WHERE (user_id = ?);";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getSurname());
+            statement.setString(3, user.getLogin());
+            statement.setString(4, user.getPassword());
+            statement.setLong(5, user.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("Can't update user", e);
+        }
+        return user;
+    }
+
+    @Override
+    public User delete(Long id) {
+        User user = get(id).get();
+        String query = "DELETE FROM " + DB_NAME
+                + ".users WHERE user_id = ?;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("Can't delete user", e);
+        }
+        return user;
+    }
+
+    @Override
+    public List<User> getAll() {
+        List<User> users = new ArrayList<>();
+        String query = "SELECT * FROM " + DB_NAME + ".users;";
+        try (PreparedStatement statement
+                     = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery(query);
+            while (resultSet.next()) {
+                User user = new User();
+                user.setId(resultSet.getLong("user_id"));
+                user.setName(resultSet.getString("name"));
+                user.setSurname(resultSet.getString("surname"));
+                user.setLogin(resultSet.getString("login"));
+                user.setToken(resultSet.getString("token"));
+                users.add(user);
+            }
+            return users;
+        } catch (SQLException e) {
+            logger.error("Get all users failed", e);
+        }
+        return users;
+    }
+
+    @Override
+    public Optional<User> login(String login, String password) throws AuthenticationException {
+        User user = null;
+        String query = "SELECT * FROM " + DB_NAME + ".users WHERE login=? AND password=?;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            statement.setString(2, password);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                user = new User();
+                user.setId(resultSet.getLong("user_id"));
+                user.setName(resultSet.getString("name"));
```

Comment:
@gkhrshch Is it possible to move this duplicated code to a separate method?
```suggestion
                private User getUserFromResultSet(ResultSet resultSet) {
                User user = new User();
                ...
                user.setName(resultSet.getString("name"));
                ....
```